### PR TITLE
SAAS-363 add redux saga

### PR DESCRIPTION
**Pull request by nicolalamacchia**

### CODE REVIEW

File: packages/platform-ui/src/index.tsx [hunk 1]
```diff
@@ -29,9 +30,11 @@ ReactDOM.render(
   <React.StrictMode>
     <ThemeContext.Provider value={light}>
       <Provider store={store}>
-        <Router>
-          <Main />
-        </Router>
+        <ConnectedRouter history={history}>
+          <Router>
```
Comment (atymchuk) [inline packages/platform-ui/src/index.tsx:34]: The router is not needed then. Pls see https://github.com/supasate/connected-react-router#step-3

File: packages/platform-ui/src/store/auth/auth.sagas.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    yield call([toast, toast.error], action.payload.message);
+  } else {
+    yield call([toast, toast.error], Messages.api.signInFailed);
+    console.error(action.payload);
+  }
+}
+
+function* authLoginSuccess(action: ReturnType<typeof authLoginAction.success>): Generator<StrictEffect, void, never> {
+  yield call([toast, toast.success], `${Messages.api.signInSucceeded} ${action.payload.email}`);
+  yield call([history, history.replace], Routes.root);
```
Comment (atymchuk) [inline packages/platform-ui/src/store/auth/auth.sagas.ts:50]: This is not cosher, pls refer to https://github.com/supasate/connected-react-router/blob/master/FAQ.md#in-redux-saga

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: packages/platform-ui/src/index.tsx (hunk 1 above)
Comment (nicolalamacchia) [reply packages/platform-ui/src/index.tsx:34]: I made it work without it, but had to downgrade `history` @ 4.10.1 (see https://github.com/ReactTraining/history/issues/804 and https://github.com/supasate/connected-react-router/issues/312)

File: packages/platform-ui/src/store/auth/auth.sagas.ts (hunk 2 above)
Comment (nicolalamacchia) [reply packages/platform-ui/src/store/auth/auth.sagas.ts]: Done

**Comment by nicolalamacchia** [comment]
e2e tests must be fixed, @yurkovychv let's have a discussion about that if needed

**Review by yurkovychv — APPROVED** [review]

File: packages/platform-ui/package.json [hunk 3]
```diff
@@ -38,23 +38,29 @@
...
     "emotion": "^10.0.27",
     "final-form": "^4.20.1",
     "google-protobuf": "^3.13.0",
     "grpc-web": "^1.2.1",
+    "history": "4.7.2",
     "react": "16.12.0",
     "react-dom": "16.12.0",
     "react-final-form": "^6.5.1",
     "react-redux": "^7.2.1",
+    "react-router": "^5.2.0",
```
Comment (atymchuk) [inline packages/platform-ui/package.json:57]: Do we need it? It's an indirect dependency of _react-router-dom_ AFAIK, so we should probably leave it out.

**Review by atymchuk — CHANGES_REQUESTED** [review]

**Review by yurkovychv — DISMISSED** [review]

File: packages/platform-ui/package.json (hunk 3 above)
Comment (nicolalamacchia) [reply packages/platform-ui/package.json]: Right, it's a mistake, removed it

**Review by atymchuk — APPROVED** [review]

**Pull request merged**